pyhafas/profile/base.py

```diff
@@ -173,6 +175,33 @@ def format_location_request(
             "meth": "LocMatch"
         }
 
+    def formatProductsFilter(self, requested_products: dict) -> dict:
+        products = self.default_products
+        for requested_product in requested_products:
+            if requested_products[requested_product]:
+                try:
+                    products.index(requested_product)
```

Comment:
If I understand this code correctly, it assumes that one product has only one bitmask. That is not always the case though.

Reply:
The calculation of the bitmask is done in lines 192-198 and not in the lines 197-183. There I loop over all requested products (`products` list) and in another loop over the bitmasks of this one product and add them to the total sum.

In this lines I create a list (`products`) first with a list of the default products of the profile. In line 180 I loop over the products requested by the client and check in line 181 if the product is requested or not. If it is requested I check in line 183 if the product is already in the `products` list. If it isn't in the list python raises a `ValueError` which is caught by the except in line 184 and then I append the requested product to the `products` list.